test/jdk/java/nio/channels/AsyncCloseAndInterrupt.java

```diff
@@ -133,11 +137,24 @@ private static void initFile() throws Exception {
             if (!fifoFile.delete())
                 throw new IOException("Cannot delete existing fifo " + fifoFile);
         }
-        Process p = Runtime.getRuntime().exec("mkfifo " + fifoFile);
-        if (p.waitFor() != 0)
-            throw new IOException("Error creating fifo");
-        new RandomAccessFile(fifoFile, "rw").close();
 
+        ProcessBuilder pb = new ProcessBuilder("mkfifo", fifoFile.toString());
+        OutputAnalyzer oa = ProcessTools.executeProcess(pb);
+        oa.waitFor();
+        if (oa.getExitValue() != 0) {
+            String stderr = oa.getStderr();
+            if (stderr.contains("mkfifo") &&
+                stderr.contains("Operation not supported")) {
+                fifoFile = null;
+                log.println("WARNING: mkfifo failed - cannot completely test FileChannels");
+                return;
+            } else {
+                throw new IOException("Error creating fifo \"" +
+                    fifoFile.getAbsolutePath() + "\"\n" +
+                    ProcessTools.getProcessLog(pb, oa));
+            }
+        }
+        new RandomAccessFile(fifoFile, "rw").close();
```

Comment:
Having you looked at using FFM to call mkinfo(2) rather launching a process to run the mkfifo tool? It's very brittle to depend on output like "Operation not supported".

Reply:
No, not yet. The test was already launching a process. I would prefer to pursue trying FFM, however, as part of a subsequent, follow-on request rather than in this one.

Reply:
> No, not yet. The test was already launching a process. I would prefer to pursue trying FFM, however, as part of a subsequent, follow-on request rather than in this one.

I think it would better to not launch a process,  it's not needed.

Reply:
Okay I replaced the `ProcessTools` section with FFM in dd7c78e28019659de95df71de2a125b61fa877d3. Should this now remove `othervm` as well?

Reply:
Never mind about `othervm` as it is required for `--enable-native-access`.

Reply:
This looks good but I can't immediately see which platform this part of the test runs on.

Reply:
If it's running on Windows, `initFile()` returns at line 155 and the `mkfifo()` method is never called.